core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/ShardsLimitAllocationDecider.java

```diff
@@ -50,66 +53,88 @@
 
     public static final String NAME = "shards_limit";
 
+    private volatile Integer clusterShardLimit;
```

Comment:
Looking at the code, it seems like it can never be null, so should it be an int instead?